src/pages/VideoDetails.jsx

```diff
@@ -1,22 +1,55 @@
+import { useParams } from 'react-router-dom';
 import { VideosWrapper } from '../components/styles/VideosWrapper.styled';
 import { PageContainer } from '../components/styles/PageContainer.styled';
 import { useContext } from 'react';
 import { GlobalContext } from '../context/GlobalContext';
 import { lightStyles } from '../components/styles/ThemeStyles.styled';
 import { darkStyles } from '../components/styles/ThemeStyles.styled';
+import { youtubeApiUrl } from '../utils/constants';
 import SuggestedVideosContainer from '../components/SuggestedVideosContainer';
 import VideoContainer from '../components/VideoContainer';
+import useFetch from '../hooks/useFetch';
 
 const VideoDetails = () => {
+  const { id } = useParams();
   const { darkMode } = useContext(GlobalContext);
   const backgroundColor = darkMode
     ? darkStyles.backgroundColor
     : lightStyles.backgroundColor;
+
+  const videoResponse = useFetch(
+    // eslint-disable-next-line no-undef
+    `${youtubeApiUrl}videos?part=snippet&id=${id}&key=${process.env.REACT_APP_API_KEY}`,
+    {},
+    id
+  );
+  const video = videoResponse.data ? videoResponse.data.items[0] : {};
```

Comment:
Maybe a better thing to do is to add the fetch hook a special prop that receives a functions that allows you to filter or do whatever you need to do with the data and that way you are sure you are getting the correct data or in the format you need it

Reply:
same comment for the suggested videos